Tools/scripts/run_flake8.py

```diff
@@ -36,10 +36,14 @@ def check(self):
             self.retcode = 1
 
     def run(self):
-        for (dirpath, dirnames, filenames) in os.walk("Tools"):
+        for (dirpath, dirnames, filenames) in os.walk("."):
             for filename in filenames:
                 if os.path.splitext(filename)[1] != ".py":
                     continue
+                if filename == 'env.py':
+                    # we are generating content into these files which
+                    # is not actually Python...
+                    continue
                 filepath = os.path.join(dirpath, filename)
                 content = open(filepath).read()
```

Comment:
This line opens a file handle but never closes it.

Can we instead `with open(filepath ) as in_file:` or `pathlib.Path(filepath).read_text()`?

Reply:
> This line opens a file handle but never closes it.
> 
> Can we instead `with open(filepath) as in_file:` or `pathlib.Path(filepath).read_text()`?

Pretty sure we're working towards nuking this script :-)

Still... I've pushed a patch for it.

Reply:
Agreed.  Thanks.  Step-by-step.